CMakeLists.txt

```diff
@@ -29,83 +37,129 @@ message( STATUS "-------- cpp2py version and git hash detection -------------")
 
 find_package(Git)
 
-# Get the Git Hash
+# get the Git Hash
 execute_process(COMMAND ${GIT_EXECUTABLE} rev-parse HEAD
- WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}"
- OUTPUT_VARIABLE   CPP2PY_GIT_HASH
- OUTPUT_STRIP_TRAILING_WHITESPACE
+  WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}"
+  OUTPUT_VARIABLE   CPP2PY_GIT_HASH
+  OUTPUT_STRIP_TRAILING_WHITESPACE
 )
 
 message(STATUS "Cpp2py version : ${CPP2PY_VERSION}")
 message(STATUS "Git hash : ${CPP2PY_GIT_HASH}")
 
-# LIBCLANG : try to set LIBCLANG_LOCATION and LIBCLANG_CXX_FLAGS automatically
-MESSAGE(STATUS "-------- LibClang detection -------------")
+#--------------------------------------------------------
+#   Libclang
+#--------------------------------------------------------
+message(STATUS "-------- LibClang detection -------------")
 
 find_package(LibClang)
 
-MESSAGE(STATUS "LibClang location: ${LIBCLANG_LOCATION}")
-MESSAGE(STATUS "LibClang additional flags: ${LIBCLANG_CXX_FLAGS}")
+message(STATUS "LibClang location: ${LIBCLANG_LOCATION}")
+message(STATUS "LibClang additional flags: ${LIBCLANG_CXX_FLAGS}")
 
 # allow user to modify
 set(LIBCLANG_CXX_FLAGS "${LIBCLANG_CXX_FLAGS}" CACHE STRING "Additional flags to be passed to libclang when parsing with clang")
 set(LIBCLANG_LOCATION "${LIBCLANG_LOCATION}" CACHE STRING "Location of the libclang library")
 
-# --- Python ---
+#--------------------------------------------------------
+#   Python and Numpy
+#--------------------------------------------------------
+find_package(Python COMPONENTS Interpreter Development NumPy)
+
+# check required python packages
+function(exec_python_script the_script output_var_name)
+  execute_process(COMMAND ${Python_EXECUTABLE} -c "${the_script}"
+    OUTPUT_VARIABLE res RESULT_VARIABLE returncode OUTPUT_STRIP_TRAILING_WHITESPACE)
+  if(NOT returncode EQUAL 0)
+    message(FATAL_ERROR "The script : ${the_script} \n did not run properly in the Python interpreter. Check your python installation.")
+  endif()
+  set(${output_var_name} ${res} PARENT_SCOPE)
+endfunction()
+
+exec_python_script("import mako.template" nulle)
+exec_python_script("import numpy" nulle)
+exec_python_script("import scipy" nulle)
+if(Build_Documentation)
+  exec_python_script("import clang.cindex" nulle)
+endif()
+
+# set variables used in the current and in dependent projects
+set(PYTHON_EXECUTABLE ${Python_EXECUTABLE} CACHE STRING "Python executable")
```

Comment:
I would suggest to consistently use `Python_EXECUTABLE` as defined by the `find_package` call.
No need to have two variables?

Reply:
No, usually we wouldn't need it. But I think it is used in the other repos (e.g. https://github.com/search?q=repo%3ATRIQS%2Fnda%20PYTHON_EXECUTABLE&type=code).
We should change does variables to `CPP2PY_PYTHON_EXECUTABLE` at some point, since that is also one of the reasons why `find_package` calls not work right now.

Reply:
Yes, it makes sense to adjust the other repos accordingly

Reply:
So I think we should first change all the other repos before removing it, right?

Reply:
I made the relevant changes for the development branches of all other repositories

Reply:
Ok, thanks! Then I will get rid of the unnecessary variables.